### PR TITLE
Add Go package registry

### LINKED ISSUE #7608: GOPROXY protocol support
If gitea could be as a goproxy server and all requested repositories could be stored on a special orgnization. That maybe a good idea.

ref: https://go.dev/ref/mod#goproxy-protocol

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #7608

This PR adds a Go package registry usable with the Go proxy protocol.

grafik

### CODE REVIEW

**Comment by silverwind** [comment]
Can you crop the SVG to a square size please?

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
New cropping is better, but I see it had no actual effect on the output image 😉

**Comment by KN4CK3R** [comment]
Yes, but the output image is already like we want it or not?

**Comment by silverwind** [comment]
Yes, output image is and was fine. I was requesting squaring of the source image, but it turns out this is not necessary after all as SVGO squares the output image correctly.

File: docs/content/doc/usage/packages/go.en-us.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+```shell
+curl --user your_username:your_password_or_token \
+     --upload-file path/to/file.zip \
+     https://gitea.example.com/api/packages/testuser/go/upload
+```
+
+If you are using 2FA or OAuth use a [personal access token]({{< relref "doc/development/api-usage.en-us.md#authentication" >}}) instead of the password.
+
+The server reponds with the following HTTP Status codes.
```
Comment (silverwind) [inline docs/content/doc/usage/packages/go.en-us.md]:
```suggestion
The server responds with the following HTTP Status codes.
```

**Comment by silverwind** [comment]
Do we need to add anything to docs regarding `GOPRIVATE` or `GOSUMDB` variables with this, so nothing leaks to public during installation of such packages?

File: docs/content/doc/usage/packages/go.en-us.md (hunk 1 above)
Comment (KN4CK3R) [reply docs/content/doc/usage/packages/go.en-us.md]: I should not have used that faulty docs template for every last package type... 😢

**Comment by KN4CK3R** [comment]
I could add a link to https://go.dev/ref/mod#private-module-privacy somewhere.

**Comment by silverwind** [comment]
Yeah I think a mention of `GOPRIVATE` is good, but no need to complicate instructions further with it.

**Review by silverwind — APPROVED** [review]

**Pull request merged**